### ISSUE TITLE
Kimi K2.6: hidden state collapses to zero between L1 and L5

### ISSUE THREAD

**Issue by ssubbotin** [ssubbotin/flash-moe#1]
## Symptom

After fixing the embed `>2GB` `pread` truncation bug (commit on `cuda-kimi`), layer 1 produces a coherent output (L2=3.03), but by layer 5 the post-layer hidden state is **exactly zero**. Every downstream layer stays at zero (`rms_norm(0) = 0`), so final logits are input-invariant.

## Repro

```bash
sudo systemctl stop vllm-qwen-coder
cd cuda_infer && make infer_kimi
KIMI_DEBUG=1 cuda_infer/infer_kimi \
  --model-dir kimi-k2.6 \
  --packed-dir kimi-k2.6/packed_experts \
  --tokens "1008" --max-tokens 1 --max-seq 16
sudo systemctl start vllm-qwen-coder
```

Per-layer prints in `kimi_forward.cuh` for layers `[0..7]` should show which layer first drops to L2=0.

## Likely suspects

- `kimi_moe_layer_forward` race / aliasing not triggered by the layer-1 manual debug path
- `cudaMemset` of `d_moe_accum`
- `d_residual` clobbered between save and combine
- Pread buffer reuse across layers (single static `kimi_host_block()` reused — could race if a pending upload hasn't completed)

## Acceptance

End-to-end inference produces non-zero, coherent logits across all 60 MoE layers, and a fixed prompt yields the same top-1 token as a reference run.

## Source notes

`~/.claude/projects/-home-sergey-flash-moe/memory/project_kimi_port.md`
